src/main/java/graphql/schema/idl/RuntimeWiring.java

```diff
@@ -307,6 +320,9 @@ public Builder type(TypeRuntimeWiring typeRuntimeWiring) {
 
             DataFetcher<?> defaultDataFetcher = typeRuntimeWiring.getDefaultDataFetcher();
             if (defaultDataFetcher != null) {
+                if (strictMode && defaultDataFetchers.containsKey(typeName)) {
+                    throw new StrictModeWiringException(format("The type %s already has a default data fetcher defined", typeName));
+                }
```

Comment:
An edge case that wasn't previously covered: if a type wiring's default datafetcher was redefined at the runtime wiring level (and not the type wiring level), previously didn't raise an error

See test below to lock in the new behaviour